modules/spring-config/src/main/java/org/mule/runtime/config/spring/MuleArtifactContext.java

```diff
@@ -177,6 +170,13 @@ protected XmlConfigurationDocumentLoader newXmlConfigurationDocumentLoader() {
     return new XmlConfigurationDocumentLoader();
   }
 
+  private XmlApplicationParser createApplicationParser() {
+    ExtensionManager extensionManager = muleContext.getExtensionManager();
+    DslResolvingContext context =
+        extensionManager != null ? DslResolvingContext.getDefault(extensionManager.getExtensions()) : null;
+    return new XmlApplicationParser(new XmlApplicationServiceRegistry(serviceRegistry, context));
```

Comment:
It won't. You need the DslResolvingContext in order to access the ExtensionModels of the plugins used by the app. This allows us to create the declaration based on the model and not in its dsl representation.